### PR TITLE
Update copyrights

### PR DESCRIPTION
**Pull request by setrofim**

### CODE REVIEW

File: dev_scripts/copyright_updater [hunk 1]
```diff
@@ -1,66 +1,215 @@
...
+    with open(filepath) as fh:
+        lines = fh.readlines()
+    for i, line in enumerate(lines):
+        if OLD_COPYRIGHT_REGEX.search(line):
+            start_line = i -4
+            break
+    try:
+        lines = lines[0:start_line] + lines[start_line + header_line_count:]
+    except UnboundLocalError:
+        import pdb; pdb.set_trace()
```
Comment (marcbonnici) [inline dev_scripts/copyright_updater]: Break point left in.

File: dev_scripts/copyright_updater [hunk 2]
```diff
@@ -1,66 +1,215 @@
...
+def get_git_year(path):
+    info = subprocess.check_output('git log -n 1 {}'.format(os.path.basename(path)),
+                                   shell = True, cwd = os.path.dirname(path))
+    if not info.strip():
+        return None
+
+    i = 1
+    while 'copyright' in info.lower():
+        info = subprocess.check_output('git log -n 1 --skip {} {}'.format(i, os.path.basename(path)),
+                                    shell = True, cwd = os.path.dirname(path))
```
Comment (marcbonnici) [inline dev_scripts/copyright_updater]: Extra spaces round ``=``

File: dev_scripts/copyright_updater [hunk 3]
```diff
@@ -1,66 +1,215 @@
...
-def Get_git_year(full_directory):
-    info = subprocess.check_output('git log -n 1 '+(os.path.basename(full_directory)),
-                                   shell = True, cwd = os.path.dirname(full_directory))
+        ret_year = '{}-{}'.format(old_year, year)
+    return (text.replace(match.group(0), updated_year_text), ret_year)
+
+
+def get_git_year(path):
+    info = subprocess.check_output('git log -n 1 {}'.format(os.path.basename(path)),
+                                   shell = True, cwd = os.path.dirname(path))
```
Comment (marcbonnici) [inline dev_scripts/copyright_updater]: Extra spaces round ``=``

File: dev_scripts/copyright_updater [hunk 4]
```diff
@@ -1,66 +1,215 @@
...
+            if exclude_path in os.path.realpath(root):
+                should_skip = True
+                break
+        if should_skip:
+            logging.info('Skipping {}'.format(root))
+            continue
 
-log_level = logging.DEBUG if args.verbose else logging.INFO
-logging.basicConfig(format='%(message)s', level=log_level)
+        logging.info('Cheking {}'.format(root))
```
Comment (marcbonnici) [inline dev_scripts/copyright_updater]: /s/Cheking/Checking

**Comment by setrofim** [comment]
Comments addressed.

**Pull request merged**